src/polyswarmd/artifacts/artifacts.py

```diff
@@ -17,6 +17,19 @@
 artifacts = Blueprint('artifacts', __name__)
 
 
+def get_size(f):
+    if f.content_length:
+        logger.debug('Content length %s', f.content_length)
+        return f.content_length
+
+    original_position = f.tell()
```

Comment:
I don't know the type of `request.files.getlist(key='file')` but if it's a regular text IO stream, it may be easier to use `fseek(*FILE, 0, SEEK_END)`:

```python
from io import SEEK_END

...

def get_size(f):
   """Return ``content_length`` or the size of an IO stream"""
    if f.content_length:
        logger.debug('Content length %s', f.content_length)
        return f.content_length
    return ff.seek(0, SEEK_END)
```

Reply:
The problem is content_length is there, but it will be 0 if not provided

Reply:
Spot on about using SEEK_END

Reply:
I added the doctest, and got it working. Requires turning on orchestration and changing settings in consul. Not ideal

Reply:
I see, I thought that `content_length` was an attribute of a different type of object that could be passed into `get_size`. This looks great!